### ISSUE TITLE
support for kafka 

### ISSUE THREAD

**Issue by mwangm** [umami-software/umami#1871]
Hi  I saw some config related with kafka and clickhouse in code,  
But  after check the doc, I can only find mysql/Postgres  section in the doc.  
is there any guide I can follow to work umami with kafka and clickhouse ?

**Comment by franciscao633** [comment]
We currently use clickhouse / kafka for our cloud hosting product, but there is no documentation at this time. We will probably only support mysql / postgres as our open-source dbs.

**Comment by luz33c** [comment]
I believe you could consider making ClickHouse and Kafka advanced options since they are present in your open-source code, but stating that the open-source version doesn't support them is not appropriate. Referring to Plausible, they emphasize the complete consistency between cloud hosting and open-source, which I find very important. 

Furthermore, ClickHouse is a mandatory option for Plausible, resulting in higher server requirements for their self-hosted solution. However, you could offer a lower-end setup with Node.js + PostgreSQL, while providing a high-end configuration featuring ClickHouse, Kafka, and PostgreSQL, which would be fantastic!

**Comment by Daniel15** [comment]
> resulting in higher server requirements for their self-hosted solution

For what it's worth, I've got a site that receives ~200k hits per month at peak (admittedly not huge, but also not small), and Clickhouse performs a lot better and uses less disk space than MySQL or PostgreSQL. I tested with Plausible and imported Google Analytics data from 2012 to now.  18.6M pageviews total. Clickhouse DB is 300MB on disk and it's using ~500MB RAM.

Any column-oriented database (not just Clickhouse) should have much better query performance for analytics use cases, since the main database queries are aggregating data by column.  Far fewer disk reads required compared to something like MySQL or PostgreSQL.
